packages/core/src/domain/trackingConsent.ts

```diff
@@ -9,25 +12,35 @@ export type TrackingConsent = (typeof TrackingConsent)[keyof typeof TrackingCons
 export interface TrackingConsentState {
   tryToInit: (trackingConsent: TrackingConsent) => void
   update: (trackingConsent: TrackingConsent) => void
-  isGranted: () => boolean
+  isGranted: (startTime?: RelativeTime) => boolean
   observable: Observable<void>
 }
 
 export function createTrackingConsentState(currentConsent?: TrackingConsent): TrackingConsentState {
   const observable = new Observable<void>()
+  const history = createValueHistory<TrackingConsent>({ expireDelay: 1000 })
+
+  if (currentConsent) {
+    history.add(currentConsent, relativeNow())
+  }
 
   return {
     tryToInit(trackingConsent: TrackingConsent) {
       if (!currentConsent) {
         currentConsent = trackingConsent
+        history.add(trackingConsent, relativeNow())
       }
     },
     update(trackingConsent: TrackingConsent) {
       currentConsent = trackingConsent
+      history.closeActive(relativeNow())
+      history.add(trackingConsent, relativeNow())
       observable.notify()
     },
-    isGranted() {
-      return currentConsent === TrackingConsent.GRANTED
+    isGranted(startTime?: RelativeTime) {
+      const value = startTime === undefined ? currentConsent : (history.find(startTime) ?? currentConsent)
```

Comment:
> [!NOTE]
> Just for the sake of types, `history.find` can return `undefined`, it should never be empty, but fallback to `currentConsent` fixes the types

Reply:
why not just?
```ts
isGranted(startTime?: RelativeTime) {
   return history.find(startTime) === TrackingConsent.GRANTED
}
```

Reply:
This would change the behavior un RUM.

Reply:
I really want to avoid changing behavior (at the moment) in other parts of the SDK. If we make this change, it will change consent behavior in RUM too.